packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -138,6 +140,58 @@ export class Rule extends Resource implements IRule {
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
 
+    if (targetProps.target) {
+      const targetStack = Stack.of(targetProps.target);
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+      if (targetStack.account !== sourceAccount) {
+        // cross-account event - strap in, this works differently than regular events!
+        // based on:
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEvents-CrossAccountEventDelivery.html
+
+        // the _actual_ target is just the event bus of the target's account
+        this.targets.push({
+          id,
+          arn: `arn:aws:events:${targetStack.region}:${targetStack.account}:event-bus/default`,
+        });
+
+        // Grant the source account permissions to the event bus of the target account.
+        // Do it in a separate stack, as it needs to be created before the rule containing the above event bus target
+        // (CloudWatch verifies whether you have permissions to the target on rule creation),
+        // but it's very likely the target stack has a dependency on the source (= this) stack
+        // (that's the case with CodePipeline, for example)
+        const stackId = `EventBusPolicy-${sourceAccount}`;
+        let eventBusPolicyStack: Stack = this.node.tryFindChild(stackId) as Stack;
```

Comment:
I wonder if we should define this stack at the app level?

Reply:
I think you're right. Changed that, and also added some logic to only have the event bus of a given account as a target once in a given rule.